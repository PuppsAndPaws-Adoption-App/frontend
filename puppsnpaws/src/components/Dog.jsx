import React from 'react';
import {Link} from 'react-router-dom';

function Dog(props) {
  


    //render the dog details
    return (
        <div className='Dog'>
        <h2 id="item-name">{props.dog.name}</h2>
        <img className="item-img" src={props.dog.image} alt={props.dog.name} />
        <Link to={`/dogs/${props.dog.id}`}>View</Link>
        </div>
    );
}

export default Dog;