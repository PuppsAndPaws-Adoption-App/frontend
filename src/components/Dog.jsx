import React from 'react';
import {Link} from 'react-router-dom';

function Dog(props) {
  
    //delete dog from db
    const clickDelete = async () => {
        await fetch(`http://localhost:3000/dogs/${props.dog.id}`, {
            method: 'DELETE'
        })
    }



    //render the dog details
    return (
        <div className='Dog'>
        <h2 id="item-name">{props.dog.name}</h2>
        <img className="item-img" src={props.dog.image} alt={props.dog.name} />
        <Link to={`/dogs/${props.dog.id}`}>View</Link>
        <button id="delete-btn" onClick={clickDelete}>Delete</button>

        </div>
    );
}

export default Dog;