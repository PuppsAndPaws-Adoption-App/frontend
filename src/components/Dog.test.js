// import { render, screen } from '@testing-library/react';
// import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
import React from "react";
import Dog from "./Dog";
import renderer from "react-test-renderer";

describe("Dog and component", () => {
  //snapshot test
  test("renders Dog component as expected", () => {
    const dog = {
      name: "Cody",
      image: "/img/Cody.jpg",
      breed: "Breed: Boston Terrier",
      gender: "Sex: Male",
      age: "Age: Two years",
      size: "Weight: 10lbs",
      description: "Brown/Black/White",
      zipCode: "75115",
    };
    //render the component
    const component = renderer.create(<Dog key={dog.id} dog={dog} />);
    //convert to json
    const componentJSON = component.toJSON();
    //expect the json component to match the snapshot
    expect(componentJSON).toMatchSnapshot();
  });

  //behavior test
  test("Dog props passed as expected", () => {
    const dog = {
      name: "Cody",
      image: "/img/Cody.jpg",
      breed: "Breed: Boston Terrier",
      gender: "Sex: Male",
      age: "Age: Two years",
      size: "Weight: 10lbs",
      description: "Brown/Black/White",
      zipCode: "75115",
    };
    //render the component
    const component = renderer.create(<Dog key={dog.id} dog={dog} />);
    //convert to tree
    const componentT = component.toTree();
    //expect the props to match a provided string
    expect(componentT.props.dog.name).toBe("Cody");
  });
});
