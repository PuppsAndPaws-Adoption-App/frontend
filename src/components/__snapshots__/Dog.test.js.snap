// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog and component renders Dog component as expected 1`] = `
<div
  className="Dog"
>
  <h2
    id="item-name"
  >
    Cody
  </h2>
  <img
    alt="Cody"
    className="item-img"
    src="/img/Cody.jpg"
  />
  <button
    id="delete-btn"
    onClick={[Function]}
  >
    Delete
  </button>
  <a
    className="edit-dog-btn"
    href="/edit-dog/undefined"
  >
    Edit this dog
  </a>
</div>
`;
